### PR TITLE
mds: remove max_mds config option

### PR DESCRIPTION
**Pull request by batrick**
It is now required that all changes to max_mds use the run-time `ceph fs set max_mds` command. The rationale for this change is that it is confusing to have a configuration for max_mds which is only observed at file system creation.

Fixes: http://tracker.ceph.com/issues/17105

### CODE REVIEW

**Comment by renhwztetecs** [comment]
:+1:

**Comment by jcsp** [comment]
Reviewed-by

File: src/common/config_opts.h [hunk 1]
```diff
@@ -530,7 +530,7 @@ OPTION(mds_inject_traceless_reply_probability, OPT_DOUBLE, 0) /* percentage
 OPTION(mds_wipe_sessions, OPT_BOOL, 0)
 OPTION(mds_wipe_ino_prealloc, OPT_BOOL, 0)
 OPTION(mds_skip_ino, OPT_INT, 0)
-OPTION(max_mds, OPT_INT, 1)
+OPTION(max_mds, OPT_INT, 1) // Obsolete and ignored! Use `ceph fs set max_mds` instead.
```
Comment (gregsfortytwo) [inline src/common/config_opts.h]: Why not just kill it entirely? I don't think unknown config options will break anything, and leaving it in is confusing.

**Comment by batrick** [comment]
@gregsfortytwo thanks, I made that change.

**Pull request merged**